plugins/aap-backend/src/providers/AapResourceEntityProvider.ts

```diff
@@ -0,0 +1,170 @@
+        providerConfig,
+        options.logger,
+        taskRunner,
+      );
+    });
+  }
+
+  private constructor(
+    config: AapConfig,
+    logger: Logger,
+    taskRunner: TaskRunner,
+  ) {
+    this.env = config.id;
+    this.baseUrl = config.baseUrl;
+    this.authorization = config.authorization;
+    this.owner = config.owner || 'AAP';
+    this.system = config.system || '';
+    this.logger = logger.child({
+      target: this.getProviderName(),
+    });
+
+    this.scheduleFn = this.createScheduleFn(taskRunner);
+  }
+
+  private createScheduleFn(taskRunner: TaskRunner): () => Promise<void> {
+    return async () => {
+      const taskId = `${this.getProviderName()}:run`;
+      return taskRunner.run({
+        id: taskId,
+        fn: async () => {
+          try {
+            await this.run();
+          } catch (error) {
+            this.logger.error(error);
+          }
+        },
+      });
+    };
+  }
+
+  getProviderName(): string {
+    return `AapResourceEntityProvider:${this.env}`;
+  }
+
+  async connect(connection: EntityProviderConnection): Promise<void> {
+    this.connection = connection;
+    await this.scheduleFn();
+  }
+
+  async run(): Promise<void> {
+    if (!this.connection) {
+      throw new Error('Not initialized');
+    }
+
+    this.logger.info(`Discovering ResourceEntities from AAP ${this.baseUrl}`);
+
+    const entities: Entity[] = [];
+
+    const jobTemplates = await listJobTemplates(
+      this.baseUrl,
+      this.authorization,
+    );
+
+    for (const template of jobTemplates) {
+      const resourceEntity: ResourceEntity =
+        this.buildApiEntityFromJobTemplate(template);
+      entities.push(resourceEntity);
+      this.logger.debug(`Discovered ResourceEntity ${template}`);
+    }
+
+    this.logger.info(`Applying the mutation with ${entities.length} entities`);
+
+    await this.connection.applyMutation({
+      type: 'full',
+      entities: entities.map(entity => ({
+        entity,
+        locationKey: this.getProviderName(),
+      })),
+    });
+  }
+
+  private buildApiEntityFromJobTemplate(template: JobTemplate): ResourceEntity {
+    const location = `${this.baseUrl}/${template.url}`;
+    const jobTemplateName = template.name.replace(/ /g, '_');
+
+    return {
+      kind: 'Resource',
+      apiVersion: 'backstage.io/v1alpha1',
+      metadata: {
+        annotations: {
+          [ANNOTATION_LOCATION]: this.getProviderName(),
+          [ANNOTATION_ORIGIN_LOCATION]: this.getProviderName(),
+        },
+        name: `${jobTemplateName}`,
+        description: `${template.description}`,
+        links: [
+          {
+            url: `${this.baseUrl}`,
+            title: 'AAP Dashboard',
+          },
+          {
+            url: `${location}`,
+            title: 'Job Template',
+          },
+        ],
+      },
+      spec: {
+        type: `${template.type}`,
+        owner: `${this.owner}`,
+        ...(this.system && { system: `${this.system}` }),
```

Comment:
```suggestion
        owner: this.owner,
        system: this.system,
```

This should be fine. Empty values should be discarded/pruned.

Reply:
Tried that but it's not working, so good to check and update

```
backend:start: 2023-07-19T06:24:08.907Z catalog warn Processor BuiltinKindsEntityProcessor threw an error while validating the entity resource:default/demo_job_template; caused by TypeError: /spec/system must NOT have fewer than 1 characters - limit: 1 type=plugin entity=resource:default/demo_job_template location=AapResourceEntityProvider:dev
```